keps/sig-node/4540-strict-cpu-reservation/README.md

```diff
@@ -0,0 +1,527 @@
+
+#### Alpha
+
+- [X] Implement the new policy option.
+- [X] Ensure proper unit tests are in place.
+
+#### Beta
+
+- [ ] Gather feedback from consumers of the new policy option.
+- [ ] Verify no major bugs reported in the previous cycle.
+
+#### GA
+
+- [ ] Allow time for feedback (1 year).
+- [ ] Make sure all risks have been addressed.
+
+### Upgrade / Downgrade Strategy
+
+The new policy option is opt-in and orthogonal to the existing ones.
+
+### Version Skew Strategy
+
+No changes needed.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+The /var/lib/kubelet/cpu\_manager\_state needs be removed when enabling or disabling the feature.
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [X] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `CPUManagerPolicyAlphaOptions`
+  - Components depending on the feature gate: `kubelet`
+- [X] Change the kubelet configuration to set a `CPUManager` policy of `static` and a `CPUManager` policy option of `strict-cpu-reservation`
+  - Will enabling / disabling the feature require downtime of the control plane? No
+  - Will enabling / disabling the feature require downtime or reprovisioning of a node?  Yes -- removing /var/lib/kubelet/cpu\_manager\_state and restarting kubelet are required.
+
+
+###### Does enabling the feature change any default behavior?
+
+Yes. Reserved CPU cores will be strictly used for system daemons and interrupt processing no longer available for workloads.
+
+The feature is only enabled when all following conditions are met:
+1. The `CPUManagerPolicyAlphaOptions` feature gate must be enabled
+2. The `static` `CPUManager` policy must be selected
+3. The new `strict-cpu-reservation` policy option must be selected
+4. The `reservedSystemCPUs` is not empty
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, the feature can be disabled by either:
+1. Disabling the `CPUManagerPolicyAlphaOptions` feature gate
+2. Removing `strict-cpu-reservation` from the list of `CPUManager` policy options
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The feature will be enabled regardless it is enabled for the first time or not.
+
+###### Are there any tests for feature enablement/disablement?
+
+- A specific e2e test will demonstrate that the default behaviour is preserved when the feature gate is disabled, or when the feature is not used (2 separate tests)
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+If the feature rollout fails, burstable and best-efforts can run on the reserved CPU cores.
+If the feature rollback fails, burstable and best-efforts can not run on the reserved CPU cores.
+In either case, existing workload will not be affected.
```

Comment:
What happens when rolling back and we don't remove `/var/lib/kubelet/cpu_manager_state` ?

Reply:
then the feature was not rolled back. I added explicit statement for this.